crates/uv-installer/src/site_packages.rs

```diff
@@ -285,99 +287,93 @@ impl SitePackages {
-                            let dependency = UnresolvedRequirementSpecification {
-                                requirement: UnresolvedRequirement::Named(Requirement::from(
-                                    dependency,
-                                )),
-                                hashes: vec![],
-                            };
-                            if seen.insert(dependency.clone()) {
-                                stack.push(dependency);
+        // Second, map all overrides to named requirements. We assume that all overrides are
+        // relevant.
+        let overrides = {
+            let mut named = Vec::with_capacity(overrides.len());
+            for requirement in overrides {
+                match &requirement.requirement {
+                    UnresolvedRequirement::Named(requirement) => {
+                        named.push(NameRequirementSpecification::from(requirement.clone()));
+                    }
+                    UnresolvedRequirement::Unnamed(requirement) => {
+                        match self.get_urls(requirement.url.verbatim.raw()).as_slice() {
+                            [] => {
+                                return Ok(SatisfiesResult::Unsatisfied(
+                                    requirement.url.verbatim.raw().to_string(),
+                                ))
+                            }
+                            [distribution] => {
+                                let requirement = uv_pep508::Requirement {
+                                    name: distribution.name().clone(),
+                                    version_or_url: Some(VersionOrUrl::Url(
+                                        requirement.url.clone(),
+                                    )),
+                                    marker: requirement.marker,
+                                    extras: requirement.extras.clone(),
+                                    origin: requirement.origin.clone(),
+                                };
+                                named.push(NameRequirementSpecification::from(Requirement::from(
+                                    requirement,
+                                )));
+                            }
+                            _ => {
+                                return Ok(SatisfiesResult::Unsatisfied(
+                                    requirement.url.verbatim.raw().to_string(),
+                                ))
                             }
                         }
                     }
                 }
-                _ => {
-                    // There are multiple installed distributions for the same package.
-                    return Ok(SatisfiesResult::Unsatisfied(entry.requirement.to_string()));
-                }
             }
-        }
+            named
+        };
 
-        Ok(SatisfiesResult::Fresh {
-            recursive_requirements: seen,
-        })
+        self.satisfies_names(&requirements, constraints, &overrides, markers)
```

Comment:
The strategy here is: map each unnamed requirement to a named requirement. Then use the "everything must have names" version of this method.